pkg/resources/validation/namespace/namespace_test.go

```
@@ -0,0 +1,306 @@
+					Name:   "anynamespace",
+					Labels: map[string]string{},
+				},
+			},
+			wantErr: false,
+			allowed: false,
+		},
+		{
+			name:        "Update things unrelated to PSA labels",
+			userName:    allowSarUser,
+			clusterName: "validcluster",
+			namespace: v1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "anynamespace",
+					Labels: map[string]string{
+						"randomkey": "randomvalue",
+					},
+					Annotations: map[string]string{
+						"annotationkey": "annotationlabel",
+					},
+					GenerateName: "abcde",
+				},
+			},
+			oldNamespace: v1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   "anynamespace",
+					Labels: map[string]string{},
+				},
+			},
+			wantErr: false,
+			allowed: true,
+		},
+	}
+
+	k8Fake := &k8testing.Fake{}
+	fakeSAR := &k8fake.FakeSubjectAccessReviews{Fake: &k8fake.FakeAuthorizationV1{Fake: k8Fake}}
+	validator := namespace.NewValidator(fakeSAR)
+
+	k8Fake.AddReactor("create", "subjectaccessreviews", func(action k8testing.Action) (handled bool, ret runtime.Object, err error) {
+		createAction := action.(k8testing.CreateActionImpl)
+		review := createAction.GetObject().(*authrizationv1.SubjectAccessReview)
+		spec := review.Spec
+
+		// simulate SAR not allowed
+		if spec.User == failSarUser {
+			review.Status.Allowed = false
+			review.Status.Reason = fmt.Sprintf("%s %s", "Can not update project PSA for:", spec.User)
+			return true, review, nil
+		}
+		review.Status.Allowed = true
+		return true, review, nil
+	})
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			request, _ := createProjectRequest(t, &tt)
+			resp, err := validator.Admit(request)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Admit() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if resp.Allowed != tt.allowed {
+				t.Errorf("Admit() got = %v, want %v", resp, tt.allowed)
+			}
+		})
+	}
+}
+
+func createProjectRequest(t *testing.T, s *test) (*admission.Request, error) {
```

this does not need to return an error because it is handled in the function